### ISSUE TITLE
New v1.12.2 Cluster Fails to Restore From Backups

### ISSUE THREAD

**Issue by nasali** [deis/deis#4797]
Hi there,

So I was attempting a restore from a backup into a newly created 5 node cluster on AWS, running version 1.12.2, and after following the http://docs.deis.io/en/latest/managing_deis/backing_up_data/ guide verbatim, I'm able to successfully execute all the steps listed until the `deisctl start platform` that simply hangs like so:
```
$ deisctl start platform
● ▴ ■
■ ● ▴ Starting Deis...
▴ ■ ●
Storage subsystem...
deis-store-monitor.service: active/running                           
deis-store-daemon.service: active/running                            
deis-store-metadata.service: active/running                          
deis-store-gateway@1.service: active/running                         
deis-store-volume.service: active/running                            
Logging subsystem...
deis-logger.service: active/running                                  
deis-logspout.service: active/running                                
Control plane...
deis-controller.service: active/running                              
deis-registry@1.service: active/running                              
deis-database.service: active/running                                
deis-builder.service: activating/start-post                          
```

I looked at the builder's journal to see what's going on, and this is what I saw:
```
$ deisctl journal builder
-- Logs begin at Mon 2015-12-14 14:06:42 UTC. --
Dec 14 16:06:23 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:23Z 1c62c9398da4 confd[600]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:23Z 1c62c9398da4 confd[600]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:26 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 72)
Dec 14 16:06:26 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:26 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:26Z 1c62c9398da4 confd[604]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:26Z 1c62c9398da4 confd[604]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:29 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 75)
Dec 14 16:06:29 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:29 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:29Z 1c62c9398da4 confd[608]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:29Z 1c62c9398da4 confd[608]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:32 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 78)
Dec 14 16:06:32 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:32 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:32Z 1c62c9398da4 confd[613]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:32Z 1c62c9398da4 confd[613]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:35 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 81)
Dec 14 16:06:35 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:35 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:35Z 1c62c9398da4 confd[617]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:35Z 1c62c9398da4 confd[617]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:38 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 84)
Dec 14 16:06:38 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:38 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:38Z 1c62c9398da4 confd[622]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:38Z 1c62c9398da4 confd[622]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:41 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 87)
Dec 14 16:06:41 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:41 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:41Z 1c62c9398da4 confd[627]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:41Z 1c62c9398da4 confd[627]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:44 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 90)
Dec 14 16:06:44 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:44 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:44Z 1c62c9398da4 confd[632]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:44Z 1c62c9398da4 confd[632]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:47 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 93)
Dec 14 16:06:47 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:47 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:47Z 1c62c9398da4 confd[637]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:47Z 1c62c9398da4 confd[637]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:50 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 96)
Dec 14 16:06:50 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:50 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:50Z 1c62c9398da4 confd[642]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:50Z 1c62c9398da4 confd[642]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:53 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 99)
Dec 14 16:06:53 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:53 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:53Z 1c62c9398da4 confd[647]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:53Z 1c62c9398da4 confd[647]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:56 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 102)
Dec 14 16:06:56 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:56 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:56Z 1c62c9398da4 confd[652]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:56Z 1c62c9398da4 confd[652]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:06:59 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 105)
Dec 14 16:06:59 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:06:59 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:06:59Z 1c62c9398da4 confd[657]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:06:59Z 1c62c9398da4 confd[657]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:07:02 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 109)
Dec 14 16:07:02 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:07:02 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:07:02Z 1c62c9398da4 confd[662]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:07:02Z 1c62c9398da4 confd[662]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:07:05 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 112)
Dec 14 16:07:05 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:07:05 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:07:05Z 1c62c9398da4 confd[666]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:07:05Z 1c62c9398da4 confd[666]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
Dec 14 16:07:08 ip-10-21-2-235.ec2.internal sh[4399]: [info] Re-trying template build. (Elapsed time: 115)
Dec 14 16:07:08 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Recoverable error: exit status 1
Dec 14 16:07:08 ip-10-21-2-235.ec2.internal sh[4399]: [debug] Output: "2015-12-14T16:07:08Z 1c62c9398da4 confd[671]: ERROR template: builder:79:49: executing \"builder\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n2015-12-14T16:07:08Z 1c62c9398da4 confd[671]: ERROR template: receiver:15:55: executing \"receiver\" at <getv \"/deis/controll...>: error calling getv: key does not exist\n"
```

So then I looked at the controller's journal and this is what I saw:
```
$ deisctl journal controller
-- Logs begin at Mon 2015-12-14 14:06:42 UTC. --
Dec 14 16:04:04 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:04:04Z 95909ec2bcb9 confd[133]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:04:04 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:04:14 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:04:14Z 95909ec2bcb9 confd[139]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:04:14 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:04:24 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:04:24Z 95909ec2bcb9 confd[145]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:04:24 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:04:34 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:04:34Z 95909ec2bcb9 confd[151]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:04:34 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:04:44 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:04:44Z 95909ec2bcb9 confd[156]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:04:44 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:04:54 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:04:54Z 95909ec2bcb9 confd[162]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:04:54 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:05:04 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:05:04Z 95909ec2bcb9 confd[168]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:05:04 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:05:14 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:05:14Z 95909ec2bcb9 confd[174]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:05:14 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:05:24 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:05:24Z 95909ec2bcb9 confd[179]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:05:24 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:05:34 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:05:34Z 95909ec2bcb9 confd[185]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:05:34 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:05:44 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:05:44Z 95909ec2bcb9 confd[190]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:05:44 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:05:54 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:05:54Z 95909ec2bcb9 confd[196]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:05:54 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:06:04 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:06:04Z 95909ec2bcb9 confd[201]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:06:04 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:06:14 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:06:14Z 95909ec2bcb9 confd[207]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:06:14 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:06:24 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:06:24Z 95909ec2bcb9 confd[213]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:06:24 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:06:34 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:06:34Z 95909ec2bcb9 confd[219]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:06:34 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:06:44 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:06:44Z 95909ec2bcb9 confd[225]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:06:44 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:06:54 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:06:54Z 95909ec2bcb9 confd[230]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:06:54 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:07:04 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:07:04Z 95909ec2bcb9 confd[236]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:07:04 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:07:14 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:07:14Z 95909ec2bcb9 confd[241]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:07:14 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
Dec 14 16:07:24 ip-10-21-2-234.ec2.internal sh[4008]: 2015-12-14T16:07:24Z 95909ec2bcb9 confd[246]: ERROR template: confd_settings.py:41:20: executing "confd_settings.py" at <getv "/deis/database...>: error calling getv: key does not exist
Dec 14 16:07:24 ip-10-21-2-234.ec2.internal sh[4008]: controller: waiting for confd to write initial templates...
```
Which sounded like it was waiting on the database to start. The database journal showed this:
```
$ deisctl journal database
-- Logs begin at Mon 2015-12-14 14:06:43 UTC. --
Dec 14 16:03:38 ip-10-21-1-127.ec2.internal sh[13453]: DETAIL: The partition being downloaded is part_00000000.tar.lzo.
Dec 14 16:03:38 ip-10-21-1-127.ec2.internal sh[13453]: HINT: The absolute S3 key is basebackups_005/base_000000050000005600000081_00000040/tar_partitions/part_00000000.tar.lzo.
Dec 14 16:03:38 ip-10-21-1-127.ec2.internal sh[13453]: STRUCTURED: time=2015-12-14T16:03:38.305184-00 pid=126
Dec 14 16:04:24 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:24 UTC LOG:  could not open usermap file "/etc/postgresql/main/pg_ident.conf": No such file or directory
Dec 14 16:04:24 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:24 UTC LOG:  database system was interrupted; last known up at 2015-11-30 22:10:20 UTC
Dec 14 16:04:24 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:24 UTC LOG:  creating missing WAL directory "pg_xlog/archive_status"
Dec 14 16:04:24 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:24 UTC LOG:  starting archive recovery
Dec 14 16:04:24 ip-10-21-1-127.ec2.internal sh[13453]: wal_e.main   INFO     MSG: starting WAL-E
Dec 14 16:04:24 ip-10-21-1-127.ec2.internal sh[13453]: DETAIL: The subcommand is "wal-fetch".
Dec 14 16:04:24 ip-10-21-1-127.ec2.internal sh[13453]: STRUCTURED: time=2015-12-14T16:04:24.481042-00 pid=231
Dec 14 16:04:24 ip-10-21-1-127.ec2.internal sh[13453]: wal_e.operator.backup INFO     MSG: begin wal restore
Dec 14 16:04:24 ip-10-21-1-127.ec2.internal sh[13453]: STRUCTURED: time=2015-12-14T16:04:24.506644-00 pid=231 action=wal-fetch key=s3://db_wal/wal_005/00000005.history.lzo prefix= seg=00000005.history state=begin
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: lzop: short read
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: wal_e.blobstore.s3.s3_util WARNING  MSG: could no longer locate object while performing wal restore
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: DETAIL: The absolute URI that could not be located is s3://db_wal/wal_005/00000005.history.lzo.
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: HINT: This can be normal when Postgres is trying to detect what timelines are available during restoration.
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: STRUCTURED: time=2015-12-14T16:04:25.069543-00 pid=231
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: wal_e.operator.backup INFO     MSG: complete wal restore
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: STRUCTURED: time=2015-12-14T16:04:25.070963-00 pid=231 action=wal-fetch key=s3://db_wal/wal_005/00000005.history.lzo prefix= seg=00000005.history state=complete
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:25 UTC FATAL:  the database system is starting up
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: wal_e.main   INFO     MSG: starting WAL-E
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: DETAIL: The subcommand is "wal-fetch".
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: STRUCTURED: time=2015-12-14T16:04:25.286461-00 pid=238
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: wal_e.operator.backup INFO     MSG: begin wal restore
Dec 14 16:04:25 ip-10-21-1-127.ec2.internal sh[13453]: STRUCTURED: time=2015-12-14T16:04:25.424418-00 pid=238 action=wal-fetch key=s3://db_wal/wal_005/000000050000005600000081.lzo prefix= seg=000000050000005600000081 state=begin
Dec 14 16:04:26 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:26 UTC FATAL:  the database system is starting up
Dec 14 16:04:27 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:27 UTC FATAL:  the database system is starting up
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:28 UTC FATAL:  the database system is starting up
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: lzop: short read
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: wal_e.blobstore.s3.s3_util WARNING  MSG: could no longer locate object while performing wal restore
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: DETAIL: The absolute URI that could not be located is s3://db_wal/wal_005/000000050000005600000081.lzo.
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: HINT: This can be normal when Postgres is trying to detect what timelines are available during restoration.
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: STRUCTURED: time=2015-12-14T16:04:28.531558-00 pid=238
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: wal_e.operator.backup INFO     MSG: complete wal restore
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: STRUCTURED: time=2015-12-14T16:04:28.534537-00 pid=238 action=wal-fetch key=s3://db_wal/wal_005/000000050000005600000081.lzo prefix= seg=000000050000005600000081 state=complete
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:28 UTC LOG:  invalid checkpoint record
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:28 UTC FATAL:  could not locate required checkpoint record
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:28 UTC HINT:  If you are not restoring from a backup, try removing the file "/var/lib/postgresql/9.3/main/backup_label".
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:28 UTC LOG:  startup process (PID 229) exited with exit code 1
Dec 14 16:04:28 ip-10-21-1-127.ec2.internal sh[13453]: 2015-12-14 16:04:28 UTC LOG:  aborting startup due to startup process failure
```

Restarting the database doesn't seem to resolve it, nor does a second restore from an earlier backup. Is there another way to resurrect the database? Or am I missing something?

Thanks a bunch!

**Comment by bacongobbler** [comment]
Do you know if you have pg_xlog available in your backup? From the sounds of it what I'm reading online about `could not locate required checkpoint record`, it sounds like your backup is corrupted.

**Comment by nasali** [comment]
@bacongobbler No, it doesn't look like there is a pg_xlog in there
```
$ find . -name "pg_xlog"
```

If I look for the file it's crying about manually, namely 000000050000005600000081.lzo, it seems to exist:
```
$ ls wal_005/
00000003.history.lzo                          000000050000005600000016.lzo                  000000050000005600000042.lzo                  00000005000000560000006E.lzo
00000004.history.lzo                          000000050000005600000017.lzo                  000000050000005600000043.lzo                  00000005000000560000006F.lzo
0000000500000055000000ED.00000028.backup.lzo  000000050000005600000018.lzo                  000000050000005600000044.lzo                  000000050000005600000070.lzo
0000000500000055000000ED.lzo                  000000050000005600000019.lzo                  000000050000005600000045.lzo                  000000050000005600000071.lzo
0000000500000055000000EE.lzo                  00000005000000560000001A.lzo                  000000050000005600000046.lzo                  000000050000005600000072.lzo
0000000500000055000000EF.lzo                  00000005000000560000001B.lzo                  000000050000005600000047.lzo                  000000050000005600000073.lzo
0000000500000055000000F0.lzo                  00000005000000560000001C.lzo                  000000050000005600000048.lzo                  000000050000005600000074.lzo
0000000500000055000000F1.lzo                  00000005000000560000001D.lzo                  000000050000005600000049.lzo                  000000050000005600000075.lzo
0000000500000055000000F2.lzo                  00000005000000560000001E.lzo                  00000005000000560000004A.lzo                  000000050000005600000076.lzo
0000000500000055000000F3.lzo                  00000005000000560000001F.lzo                  00000005000000560000004B.lzo                  000000050000005600000077.lzo
0000000500000055000000F4.lzo                  000000050000005600000020.lzo                  00000005000000560000004C.lzo                  000000050000005600000078.lzo
0000000500000055000000F5.lzo                  000000050000005600000021.lzo                  00000005000000560000004D.lzo                  000000050000005600000079.lzo
0000000500000055000000F6.lzo                  000000050000005600000022.lzo                  00000005000000560000004E.lzo                  00000005000000560000007A.lzo
0000000500000055000000F7.lzo                  000000050000005600000023.lzo                  00000005000000560000004F.lzo                  00000005000000560000007B.lzo
0000000500000055000000F8.lzo                  000000050000005600000024.lzo                  000000050000005600000050.lzo                  00000005000000560000007C.lzo
0000000500000055000000F9.lzo                  000000050000005600000025.lzo                  000000050000005600000051.lzo                  00000005000000560000007D.lzo
0000000500000055000000FA.lzo                  000000050000005600000026.lzo                  000000050000005600000052.lzo                  00000005000000560000007E.lzo
0000000500000055000000FB.lzo                  000000050000005600000027.lzo                  000000050000005600000053.lzo                  00000005000000560000007F.lzo
0000000500000055000000FC.lzo                  000000050000005600000028.lzo                  000000050000005600000054.lzo                  000000050000005600000080.lzo
0000000500000055000000FD.lzo                  000000050000005600000029.lzo                  000000050000005600000055.lzo                  000000050000005600000081.00000028.backup.lzo
0000000500000055000000FE.lzo                  00000005000000560000002A.lzo                  000000050000005600000056.lzo                  000000050000005600000081.lzo
0000000500000055000000FF.lzo                  00000005000000560000002B.lzo                  000000050000005600000057.lzo                  000000050000005600000082.lzo
000000050000005600000000.lzo                  00000005000000560000002C.lzo                  000000050000005600000058.lzo                  000000050000005600000083.lzo
000000050000005600000001.lzo                  00000005000000560000002D.lzo                  000000050000005600000059.lzo                  000000050000005600000084.lzo
000000050000005600000002.lzo                  00000005000000560000002E.lzo                  00000005000000560000005A.lzo                  000000050000005600000085.lzo
000000050000005600000003.lzo                  00000005000000560000002F.lzo                  00000005000000560000005B.lzo                  000000050000005600000086.lzo
000000050000005600000004.lzo                  000000050000005600000030.lzo                  00000005000000560000005C.00000028.backup.lzo  000000050000005600000087.lzo
000000050000005600000005.lzo                  000000050000005600000031.lzo                  00000005000000560000005C.lzo                  000000050000005600000088.lzo
000000050000005600000006.lzo                  000000050000005600000032.lzo                  00000005000000560000005D.lzo                  000000050000005600000089.lzo
000000050000005600000007.lzo                  000000050000005600000033.lzo                  00000005000000560000005E.lzo                  00000005000000560000008A.lzo
000000050000005600000008.lzo                  000000050000005600000034.lzo                  00000005000000560000005F.lzo                  00000005000000560000008B.lzo
000000050000005600000009.lzo                  000000050000005600000035.lzo                  000000050000005600000060.lzo                  00000005000000560000008C.lzo
00000005000000560000000A.lzo                  000000050000005600000036.lzo                  000000050000005600000061.lzo                  00000005000000560000008D.lzo
00000005000000560000000B.lzo                  000000050000005600000037.00000028.backup.lzo  000000050000005600000062.lzo                  00000005000000560000008E.lzo
00000005000000560000000C.lzo                  000000050000005600000037.lzo                  000000050000005600000063.lzo                  00000005000000560000008F.lzo
00000005000000560000000D.lzo                  000000050000005600000038.lzo                  000000050000005600000064.lzo                  000000050000005600000090.lzo
00000005000000560000000E.lzo                  000000050000005600000039.lzo                  000000050000005600000065.lzo                  000000050000005600000091.lzo
00000005000000560000000F.lzo                  00000005000000560000003A.lzo                  000000050000005600000066.lzo                  000000050000005600000092.lzo
000000050000005600000010.lzo                  00000005000000560000003B.lzo                  000000050000005600000067.lzo                  000000050000005600000093.lzo
000000050000005600000011.lzo                  00000005000000560000003C.lzo                  000000050000005600000068.lzo                  000000050000005600000094.lzo
000000050000005600000012.00000028.backup.lzo  00000005000000560000003D.lzo                  000000050000005600000069.lzo                  000000050000005600000095.lzo
000000050000005600000012.lzo                  00000005000000560000003E.lzo                  00000005000000560000006A.lzo                  000000050000005600000096.lzo
000000050000005600000013.lzo                  00000005000000560000003F.lzo                  00000005000000560000006B.lzo                  000000050000005600000097.lzo
000000050000005600000014.lzo                  000000050000005600000040.lzo                  00000005000000560000006C.lzo                  00000005.history.lzo
000000050000005600000015.lzo                  000000050000005600000041.lzo                  00000005000000560000006D.lzo
```

I've attempted to restore using three different backups, and they all fail the same way.

**Comment by bacongobbler** [comment]
This is not something we'll be able to get to for the LTS release (#4776) and the architecture in deis v2 has dramatically changed such that this is no longer relevant so I'm closing this. If you're able to reproduce this issue and produce a way we can fix this then that would be great, otherwise closing this as fixed in v2.

**Comment by rstacruz** [comment]
i'm getting the same, but with this in `deisctl journal database`:

Jan 25 11:46:27 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: 2016-01-25 11:46:27 UTC FATAL:  the database system is starting up
Jan 25 11:46:28 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: wal_e.main   INFO     MSG: starting WAL-E
Jan 25 11:46:28 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: DETAIL: The subcommand is "wal-fetch".
Jan 25 11:46:28 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: STRUCTURED: time=2016-01-25T11:46:28.409880-00 pid=4138
Jan 25 11:46:28 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: 2016-01-25 11:46:28 UTC FATAL:  the database system is starting up
Jan 25 11:46:29 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: wal_e.operator.backup INFO     MSG: begin wal restore
Jan 25 11:46:29 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: STRUCTURED: time=2016-01-25T11:46:29.035808-00 pid=4138 action=wal-fetch key=s3://db_wal/wal_005/0000000300000010000000A9.lzo prefix= seg=0000000300000010000000A9 state=begin
Jan 25 11:46:30 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: 2016-01-25 11:46:30 UTC FATAL:  the database system is starting up
Jan 25 11:46:31 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: 2016-01-25 11:46:31 UTC FATAL:  the database system is starting up
Jan 25 11:46:32 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: 2016-01-25 11:46:32 UTC FATAL:  the database system is starting up
Jan 25 11:46:33 ip-10-21-1-69.ap-southeast-2.compute.internal sh[6467]: 2016-01-25 11:46:33 UTC FATAL:  the database system is starting up
